e2e/e2e.go

```diff
@@ -314,7 +317,14 @@ func main() { //nolint:funlen,gocyclo // Why: there are no reusable parts to ext
 		go func(wg *sync.WaitGroup) {
 			defer wg.Done()
 			log.Info().Msg("Running devconfig in background")
-			runDevconfig(ctx)
+			if err := runDevconfig(ctx); err != nil {
+				// Call cancel to hopefully communicate a signal back to other currently running commands to stop
+				// doing what they're doing. If we just exit (implicitly via log.Fatal) they likely will continue
```

Comment:
It looks like this is using a third-party logger, but it looks like `Fatal` behavior [is the same as the stdlib](https://github.com/rs/zerolog/blob/9070d49a1a4154c819e9c5ae6c923f3cc916d959/log.go#L368).

Since this is calling `os.Exit`, a `cancel` call likely won't have enough time to finish working. Is there a reason you can't just return from this function?

Reply:
The entirety of this "background runner" is written extremely poorly is the thing. I didn't have a ton of time to sink into fixing this. You're probably right though. I think this needs a little more TLC to work correctly.

Reply:
There are a few ways in which we could consistently guarantee that a canceled context coming from a spawned (non-main) go-routine ends up killing all spawned processes (from os.Exec). It's going to require some sort of coordination framework though. Go really doesn't make this easy unfortunately